### PR TITLE
Reduce string allocations at startup

### PR DESCRIPTION
**Pull request by Bond-009**
Did some cleanup while I was there.

### CODE REVIEW

File: Emby.Server.Implementations/Services/ServicePath.cs [hunk 1]
```diff
@@ -403,46 +380,39 @@ public bool IsMatch(string httpMethod, string[] withPathInfoParts, out int wildc
...
 
-        private static bool LiteralsEqual(string str1, string str2)
-        {
-            // Most cases
-            if (string.Equals(str1, str2, StringComparison.OrdinalIgnoreCase))
-            {
-                return true;
-            }
-
-            // Handle turkish i
```
Comment (JustAMan) [inline Emby.Server.Implementations/Services/ServicePath.cs]: are you sure current case of using `StringComparison.InvariantCultureIgnoreCase` covers "turkish i" case?

File: Emby.Server.Implementations/Services/ServicePath.cs [hunk 2]
```diff
@@ -538,7 +509,7 @@ public object CreateRequest(string pathInfo, Dictionary<string, string> queryStr
                     pathIx++;
                 }
 
-                requestKeyValuesMap[propertyNameOnRequest] = value;
+                requestKeyValuesMap[variableName] = value;
```
Comment (JustAMan) [inline Emby.Server.Implementations/Services/ServicePath.cs]: won't this be a problem that previously `requestKeyValuesMap` had lower-case keys, and now they could be arbitrary-cased?

File: Emby.Server.Implementations/Services/ServicePath.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Services/ServicePath.cs]: Afaik no, but the current implementation doesn't either (at least in my tests).

File: Emby.Server.Implementations/Services/ServicePath.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Services/ServicePath.cs]: No, I changed the dict to ignorecase.

Comment (JustAMan) [reply Emby.Server.Implementations/Services/ServicePath.cs]: I mean, someone might rely on serving _out_ those keys, and they were served lower-case before while now they're arbitrary. It's a minor concern anyway...

**Pull request merged**